slackroll

```diff
@@ -1755,24 +1835,13 @@ try:
 
     operation = sys.argv[1]
     args = sys.argv[2:]
+    verify_operation_and_args(operation, op_num_args, args)
 
-    # Verify number of arguments
-    if operation not in op_num_args:
-        # Find the operations that the user may be meaning.
-        given_words = operation.split('-')
-        command_words = [tuple(x.split('-')) for x in op_num_args.keys()]
-        distances = dict((x, words_to_words_distance(given_words, x)) for x in command_words)
-        min_distance = min(distances.values())
-        best_matches = [x for x in distances if distances[x] == min_distance]
-        best_matches.sort()
-
-        # Print the error and the list.
-        sys.stderr.write('ERROR: no operation called "%s"\n' % (operation, ))
-        sys.stderr.write('Use the "help" operation to get a list.\n')
-        sys.stderr.write('Did you mean %s?\n' % ' or '.join('"%s"' % ('-'.join(x), ) for x in best_matches))
-        sys.exit(slackroll_exit_failure)
-    else:
-        verify_num_args(operation, op_num_args[operation], args)
+    if operation == 'batch':
+        operation = args[0]
+        slackroll_batch_mode = True
+        args = args[1:]
+        verify_operation_and_args(operation, op_num_args, args)
```

Comment:
You're not removing "batch" before the second verification, so "slackroll batch batch batch", "slackroll batch batch list-transient", etc, are legal now and do nothing. Use "del op_num_args['batch']" before the second verification to forbid it.

Reply:
Ah yes nice catch. Updated now.